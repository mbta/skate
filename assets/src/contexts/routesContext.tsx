import React, { createContext, ReactElement, useContext } from "react"
import { Route, RouteId } from "../schedule"

const RoutesContext = createContext<Route[] | null>(null)

export const useRoute = (routeId: RouteId | null | undefined): Route | null => {
  const routes: Route[] | null = useContext(RoutesContext)
  return (routes || []).find((route) => route.id === routeId) || null
}

export const RoutesProvider = ({
  routes,
  children,
}: {
  routes: Route[] | null
  children: ReactElement<HTMLElement>
}) => {
  return (
    <RoutesContext.Provider value={routes}>{children}</RoutesContext.Provider>
  )
}

export default RoutesContext
