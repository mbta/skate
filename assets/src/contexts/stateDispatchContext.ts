import { createContext } from "react"
import { Dispatch, initialState, State } from "../state"

type StateDispatchContextData = [State, Dispatch]

// tslint:disable-next-line: no-empty
const noopDispatch: Dispatch = () => {}
const StateDispatchContext = createContext([
  initialState,
  noopDispatch,
] as StateDispatchContextData)

export default StateDispatchContext
