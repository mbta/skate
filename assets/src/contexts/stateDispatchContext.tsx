import React, { createContext, ReactElement } from "react"
import { Dispatch, initialState, State } from "../state"

export type StateDispatchContextData = [State, Dispatch]

// tslint:disable-next-line: no-empty
const noopDispatch: Dispatch = () => {}
export const StateDispatchContext = createContext([
  initialState,
  noopDispatch,
] as StateDispatchContextData)

export const StateDispatchProvider = ({
  state,
  dispatch,
  children,
}: {
  state: State
  dispatch: Dispatch
  children: ReactElement<HTMLElement>
}) => {
  return (
    <StateDispatchContext.Provider value={[state, dispatch]}>
      {children}
    </StateDispatchContext.Provider>
  )
}
