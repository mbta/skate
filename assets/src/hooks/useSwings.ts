import { useEffect, useState } from "react"
import { fetchSwings } from "../api"
import { Swing } from "../schedule.d"

const useSwings = (): Swing[] | null => {
  const [swings, setSwings] = useState<Swing[] | null>(null)
  useEffect(() => {
    fetchSwings().then((newSwings: Swing[] | null) => setSwings(newSwings))
  }, [])
  return swings
}

export default useSwings
