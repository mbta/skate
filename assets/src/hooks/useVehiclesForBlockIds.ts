import { Socket } from "phoenix"
import { nullableParser } from "../api"
import {
  VehicleOrGhostData,
  vehicleOrGhostFromData,
} from "../models/vehicleData"
import { VehicleOrGhost } from "../realtime"
import { BlockId } from "../schedule"
import { useChannel } from "./useChannel"

const parser = (data: VehicleOrGhostData[]): VehicleOrGhost[] =>
  data.map(vehicleOrGhostFromData)

const useVehiclesForBlockIds = (
  socket: Socket | undefined,
  blockIds: BlockId[]
): VehicleOrGhost[] | null | undefined => {
  const topic: string | null =
    blockIds.length > 0 ? `vehicles:block_ids:${blockIds.join(",")}` : null

  return useChannel<VehicleOrGhost[] | null | undefined>({
    socket,
    topic,
    event: "vehicles",
    parser: nullableParser(parser),
    loadingState: undefined,
  })
}

export default useVehiclesForBlockIds
