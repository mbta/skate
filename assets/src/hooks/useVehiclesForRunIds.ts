import { Socket } from "phoenix"
import { nullableParser } from "../api"
import {
  VehicleOrGhostData,
  vehicleOrGhostFromData,
} from "../models/vehicleData"
import { VehicleOrGhost } from "../realtime"
import { RunId } from "../realtime"
import { useChannel } from "./useChannel"

const parser = (data: VehicleOrGhostData[]): VehicleOrGhost[] =>
  data.map(vehicleOrGhostFromData)

const useVehiclesForRunIds = (
  socket: Socket | undefined,
  runIds: RunId[]
): VehicleOrGhost[] | null | undefined => {
  const topic: string | null =
    runIds.length > 0 ? `vehicles:run_ids:${runIds.join(",")}` : null

  return useChannel<VehicleOrGhost[] | null | undefined>({
    socket,
    topic,
    event: "vehicles",
    parser: nullableParser(parser),
    loadingState: undefined,
  })
}

export default useVehiclesForRunIds
