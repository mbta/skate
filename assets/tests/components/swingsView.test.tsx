import { mount } from "enzyme"
import React from "react"
import renderer from "react-test-renderer"
import ghostFactory from "../factories/ghost"
import vehicleFactory from "../factories/vehicle"
import swingFactory from "../factories/swing"
import SwingsView from "../../src/components/swingsView"
import { RoutesProvider } from "../../src/contexts/routesContext"
import { StateDispatchProvider } from "../../src/contexts/stateDispatchContext"
import useSwings from "../../src/hooks/useSwings"
import useVehiclesForRunIds from "../../src/hooks/useVehiclesForRunIds"
import useVehiclesForBlockIds from "../../src/hooks/useVehiclesForBlockIds"
import { Route, Swing } from "../../src/schedule"
import { initialState, selectVehicle, toggleSwingsView } from "../../src/state"
import { Vehicle, Ghost, VehicleOrGhost } from "../../src/realtime"
import * as dateTime from "../../src/util/dateTime"

jest.mock("../../src/hooks/useSwings", () => ({
  __esModule: true,
  default: jest.fn(),
}))

jest.mock("../../src/hooks/useVehiclesForRunIds", () => ({
  __esModule: true,
  default: jest.fn(),
}))

jest.mock("../../src/hooks/useVehiclesForBlockIds", () => ({
  __esModule: true,
  default: jest.fn(),
}))

jest.spyOn(dateTime, "now").mockImplementation(() => {
  return new Date(18000 * 1000)
})

const vehicle: Vehicle = vehicleFactory.build({
  runId: "123-456",
  blockId: "A12-34",
})

const ghost: Ghost = ghostFactory.build({ runId: "124-456" })

const routes: Route[] = [
  {
    id: "1",
    name: "Name 1",
    directionNames: {
      0: "Someplace",
      1: "Some Otherplace",
    },
  },
  {
    id: "2",
    name: "Name 3",
    directionNames: {
      0: "Someplace",
      1: "Some Otherplace",
    },
  },
  {
    id: "3",
    name: "Name 3",
    directionNames: {
      0: "Someplace",
      1: "Some Otherplace",
    },
  },
]

describe("SwingsView", () => {
  test("renders loading message", () => {
    ;(useSwings as jest.Mock).mockImplementationOnce(() => null)
    const tree = renderer
      .create(
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("omits swings more than 10 minutes in the past", () => {
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build(),
    ])

    const tree = renderer
      .create(
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("includes swings less than 10 minutes in the past", () => {
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build({ time: 17700 }),
    ])

    const tree = renderer
      .create(
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("renders future swings, active and inactive", () => {
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build({ time: 19000 }),
      swingFactory.build({
        blockId: "B12-34",
        fromRouteId: "2",
        fromRunId: "124-456",
        fromTripId: "1235",
        toRouteId: "2",
        toRunId: "124-789",
        toTripId: "5679",
        time: 20000,
      }),
      swingFactory.build({
        blockId: "C12-34",
        fromRouteId: "3",
        fromRunId: "125-456",
        fromTripId: "1236",
        toRouteId: "3",
        toRunId: "125-789",
        toTripId: "5680",
        time: 21000,
      }),
    ])
    ;(useVehiclesForRunIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle, ghost]
    )
    ;(useVehiclesForBlockIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )

    const tree = renderer
      .create(
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("ignores vehicles without run ID (for linking to VPP)", () => {
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build({ time: 19000 }),
    ])
    ;(useVehiclesForRunIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [{ ...vehicle, runId: null }]
    )
    ;(useVehiclesForBlockIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [{ ...vehicle, runId: null }]
    )

    const tree = renderer
      .create(
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("opens VPP when clicking an active swing-off and sends Fullstory event", () => {
    const originalFS = window.FS
    const originalUsername = window.username
    window.FS = { event: jest.fn(), identify: jest.fn() }
    window.username = "username"

    afterEach(() => {
      window.FS = originalFS
      window.username = originalUsername
    })
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build({ time: 19000 }),
    ])
    ;(useVehiclesForRunIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )
    ;(useVehiclesForBlockIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )

    const dispatch = jest.fn()
    const wrapper = mount(
      <StateDispatchProvider state={initialState} dispatch={dispatch}>
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      </StateDispatchProvider>
    )

    wrapper.find("a").first().simulate("click")
    expect(dispatch).toHaveBeenCalledWith(selectVehicle("v1"))
    expect(window.FS!.event).toHaveBeenCalledWith(
      "Clicked on swing-off from swings view"
    )
  })

  test("opens VPP when clicking an active swing-on and sends Fullstory event", () => {
    const originalFS = window.FS
    const originalUsername = window.username
    window.FS = { event: jest.fn(), identify: jest.fn() }
    window.username = "username"

    afterEach(() => {
      window.FS = originalFS
      window.username = originalUsername
    })
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build({
        fromRunId: "123-789",
        toRunId: "123-456",
        time: 19000,
      }),
    ])
    ;(useVehiclesForRunIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )
    ;(useVehiclesForBlockIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )

    const dispatch = jest.fn()
    const wrapper = mount(
      <StateDispatchProvider state={initialState} dispatch={dispatch}>
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      </StateDispatchProvider>
    )

    wrapper.find("a").first().simulate("click")
    expect(dispatch).toHaveBeenCalledWith(selectVehicle("v1"))
    expect(window.FS!.event).toHaveBeenCalledWith(
      "Clicked on swing-on from swings view"
    )
  })

  test("links to both swing-on and swing-off if both are active", () => {
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build({ time: 19000 }),
    ])

    const vehicle2 = vehicleFactory.build({ runId: "123-789" })
    ;(useVehiclesForRunIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle, vehicle2]
    )
    ;(useVehiclesForBlockIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )

    const tree = renderer.create(
      <StateDispatchProvider state={initialState} dispatch={jest.fn()}>
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      </StateDispatchProvider>
    )
    expect(tree).toMatchSnapshot()
  })

  test("can close the swings view", () => {
    ;(useSwings as jest.Mock).mockImplementationOnce((): Swing[] => [
      swingFactory.build({ time: 19000 }),
    ])
    ;(useVehiclesForRunIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )
    ;(useVehiclesForBlockIds as jest.Mock).mockImplementationOnce(
      (): VehicleOrGhost[] => [vehicle]
    )

    const dispatch = jest.fn()
    const wrapper = mount(
      <StateDispatchProvider state={initialState} dispatch={dispatch}>
        <RoutesProvider routes={routes}>
          <SwingsView />
        </RoutesProvider>
      </StateDispatchProvider>
    )

    wrapper.find(".m-close-button").first().simulate("click")
    expect(dispatch).toHaveBeenCalledWith(toggleSwingsView())
  })
})
