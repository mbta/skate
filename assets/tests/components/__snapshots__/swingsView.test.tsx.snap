// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwingsView ignores vehicles without run ID (for linking to VPP) 1`] = `
<div
  className="m-swings-view"
  id="m-swings-view"
>
  <button
    className="m-close-button"
    onClick={[Function]}
  >
    <span
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "SVG",
        }
      }
    />
  </button>
  <div
    className="m-swings-view__header"
  >
    Swings view
  </div>
  <div
    className="m-swings-view__description"
  >
    Upcoming swings on your selected routes
  </div>
  <div
    className="m-swings-view__table-container"
  >
    <table
      className="m-swings-view__table"
    >
      <thead
        className="m-swings-view__table-header"
      >
        <tr>
          <th
            className="m-swings-view__table-header-cell"
          >
            <div>
              Swing On
              <div
                className="m-swings-view__table-header-cell-subheaders"
              >
                <div
                  className="m-swings-view__table-header-cell-subheader"
                >
                  Time
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-on"
          >
            Swing On
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-off"
          >
            Swing Off
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
              <div
                className="m-swings-view__table-header-cell-route-subheader"
              >
                Route
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell"
          >
            Vehicle
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="m-swings-view__show-past m-swings-view__show-past-disabled"
            colSpan={4}
            onClick={[Function]}
          >
            <span
              className="m-swings-view__show-past-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Show past swings
          </th>
        </tr>
        <tr
          className="m-swings-view__table-row-inactive"
        >
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              5:16 AM
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              789
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              456
              <div
                className="m-swings-view__route-pill"
              >
                <div
                  className="m-swings-view__route"
                >
                  Name 1
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              v1-label
            </div>
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`SwingsView includes swings less than 15 minutes in the past 1`] = `
<div
  className="m-swings-view"
  id="m-swings-view"
>
  <button
    className="m-close-button"
    onClick={[Function]}
  >
    <span
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "SVG",
        }
      }
    />
  </button>
  <div
    className="m-swings-view__header"
  >
    Swings view
  </div>
  <div
    className="m-swings-view__description"
  >
    Upcoming swings on your selected routes
  </div>
  <div
    className="m-swings-view__table-container"
  >
    <table
      className="m-swings-view__table"
    >
      <thead
        className="m-swings-view__table-header"
      >
        <tr>
          <th
            className="m-swings-view__table-header-cell"
          >
            <div>
              Swing On
              <div
                className="m-swings-view__table-header-cell-subheaders"
              >
                <div
                  className="m-swings-view__table-header-cell-subheader"
                >
                  Time
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-on"
          >
            Swing On
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-off"
          >
            Swing Off
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
              <div
                className="m-swings-view__table-header-cell-route-subheader"
              >
                Route
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell"
          >
            Vehicle
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="m-swings-view__show-past m-swings-view__show-past-disabled"
            colSpan={4}
            onClick={[Function]}
          >
            <span
              className="m-swings-view__show-past-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Show past swings
          </th>
        </tr>
        <tr
          className="m-swings-view__table-row-inactive"
        >
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              4:45 AM
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              789
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              456
              <div
                className="m-swings-view__route-pill"
              >
                <div
                  className="m-swings-view__route"
                >
                  Name 1
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            />
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`SwingsView links to both swing-on and swing-off if both are active 1`] = `
<div
  className="m-swings-view"
  id="m-swings-view"
>
  <button
    className="m-close-button"
    onClick={[Function]}
  >
    <span
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "SVG",
        }
      }
    />
  </button>
  <div
    className="m-swings-view__header"
  >
    Swings view
  </div>
  <div
    className="m-swings-view__description"
  >
    Upcoming swings on your selected routes
  </div>
  <div
    className="m-swings-view__table-container"
  >
    <table
      className="m-swings-view__table"
    >
      <thead
        className="m-swings-view__table-header"
      >
        <tr>
          <th
            className="m-swings-view__table-header-cell"
          >
            <div>
              Swing On
              <div
                className="m-swings-view__table-header-cell-subheaders"
              >
                <div
                  className="m-swings-view__table-header-cell-subheader"
                >
                  Time
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-on"
          >
            Swing On
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-off"
          >
            Swing Off
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
              <div
                className="m-swings-view__table-header-cell-route-subheader"
              >
                Route
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell"
          >
            Vehicle
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="m-swings-view__show-past m-swings-view__show-past-disabled"
            colSpan={4}
            onClick={[Function]}
          >
            <span
              className="m-swings-view__show-past-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Show past swings
          </th>
        </tr>
        <tr
          className="m-swings-view__table-row-active"
        >
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              5:16 AM
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              <span
                className="m-swings-view__run-icon m-swings-view__run-icon-arrow"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SVG",
                  }
                }
              />
              <a
                onClick={[Function]}
              >
                789
              </a>
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              <span
                className="m-swings-view__run-icon m-swings-view__run-icon-arrow"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SVG",
                  }
                }
              />
              <a
                onClick={[Function]}
              >
                456
              </a>
              <div
                className="m-swings-view__route-pill"
              >
                <div
                  className="m-swings-view__route"
                >
                  Name 1
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              v1-label
            </div>
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`SwingsView omits swings more than 15 minutes in the past 1`] = `
<div
  className="m-swings-view"
  id="m-swings-view"
>
  <button
    className="m-close-button"
    onClick={[Function]}
  >
    <span
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "SVG",
        }
      }
    />
  </button>
  <div
    className="m-swings-view__header"
  >
    Swings view
  </div>
  <div
    className="m-swings-view__description"
  >
    Upcoming swings on your selected routes
  </div>
  <div
    className="m-swings-view__table-container"
  >
    <table
      className="m-swings-view__table"
    >
      <thead
        className="m-swings-view__table-header"
      >
        <tr>
          <th
            className="m-swings-view__table-header-cell"
          >
            <div>
              Swing On
              <div
                className="m-swings-view__table-header-cell-subheaders"
              >
                <div
                  className="m-swings-view__table-header-cell-subheader"
                >
                  Time
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-on"
          >
            Swing On
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-off"
          >
            Swing Off
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
              <div
                className="m-swings-view__table-header-cell-route-subheader"
              >
                Route
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell"
          >
            Vehicle
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="m-swings-view__show-past m-swings-view__show-past-disabled"
            colSpan={4}
            onClick={[Function]}
          >
            <span
              className="m-swings-view__show-past-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Show past swings
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`SwingsView renders future swings, active and inactive 1`] = `
<div
  className="m-swings-view"
  id="m-swings-view"
>
  <button
    className="m-close-button"
    onClick={[Function]}
  >
    <span
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "SVG",
        }
      }
    />
  </button>
  <div
    className="m-swings-view__header"
  >
    Swings view
  </div>
  <div
    className="m-swings-view__description"
  >
    Upcoming swings on your selected routes
  </div>
  <div
    className="m-swings-view__table-container"
  >
    <table
      className="m-swings-view__table"
    >
      <thead
        className="m-swings-view__table-header"
      >
        <tr>
          <th
            className="m-swings-view__table-header-cell"
          >
            <div>
              Swing On
              <div
                className="m-swings-view__table-header-cell-subheaders"
              >
                <div
                  className="m-swings-view__table-header-cell-subheader"
                >
                  Time
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-on"
          >
            Swing On
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell m-swings-view__table-header-cell-swing-off"
          >
            Swing Off
            <div
              className="m-swings-view__table-header-cell-subheaders"
            >
              <div
                className="m-swings-view__table-header-cell-subheader"
              >
                Run
              </div>
              <div
                className="m-swings-view__table-header-cell-route-subheader"
              >
                Route
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-header-cell"
          >
            Vehicle
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            className="m-swings-view__show-past m-swings-view__show-past-disabled"
            colSpan={4}
            onClick={[Function]}
          >
            <span
              className="m-swings-view__show-past-icon"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "SVG",
                }
              }
            />
            Show past swings
          </th>
        </tr>
        <tr
          className="m-swings-view__table-row-active"
        >
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              5:16 AM
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              789
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              <span
                className="m-swings-view__run-icon m-swings-view__run-icon-arrow"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SVG",
                  }
                }
              />
              <a
                onClick={[Function]}
              >
                456
              </a>
              <div
                className="m-swings-view__route-pill"
              >
                <div
                  className="m-swings-view__route"
                >
                  Name 1
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              v1-label
            </div>
          </th>
        </tr>
        <tr
          className="m-swings-view__table-row-active"
        >
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              5:33 AM
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              789
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              <span
                className="m-swings-view__run-icon m-swings-view__run-icon-ghost"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "SVG",
                  }
                }
              />
              <a
                onClick={[Function]}
              >
                456
              </a>
              <div
                className="m-swings-view__route-pill"
              >
                <div
                  className="m-swings-view__route"
                >
                  Name 3
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            />
          </th>
        </tr>
        <tr
          className="m-swings-view__table-row-inactive"
        >
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              5:50 AM
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              789
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            >
              456
              <div
                className="m-swings-view__route-pill"
              >
                <div
                  className="m-swings-view__route"
                >
                  Name 3
                </div>
              </div>
            </div>
          </th>
          <th
            className="m-swings-view__table-cell"
          >
            <div
              className="m-swings-view__table-cell-contents"
            />
          </th>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`SwingsView renders loading message 1`] = `
<div
  className="m-swings-view"
  id="m-swings-view"
>
  <button
    className="m-close-button"
    onClick={[Function]}
  >
    <span
      className=""
      dangerouslySetInnerHTML={
        Object {
          "__html": "SVG",
        }
      }
    />
  </button>
  <div
    className="m-swings-view__header"
  >
    Swings view
  </div>
  <div
    className="m-swings-view__description"
  >
    Upcoming swings on your selected routes
  </div>
  loading...
</div>
`;
